### PR TITLE
Editor: feat channel mentions

### LINKED ISSUE #124: Editor: support unique channel names and slash mentions for channels
Needs to support

- ex `/founders` slash command search in channels dropdown, including the unique channel slug there
- `/` dropdown in editor when mentioning a channel
- syntax highlighting for channels in casts

### PR DESCRIPTION
**Pull request by davidfurlong**
## Change Summary

- updates the api to support neynar full channel types, versioning it for backwards compat
- fixes #124 

## Merge Checklist

- [x] PR has a changeset
- [x] PR includes documentation if necessary
- [x] PR updates the [rich-embed examples](https://github.com/mod-protocol/mod/blob/main/examples/nextjs-shadcn/src/app/embeds.tsx) if necessary
- [ ] includes a parallel PR for [Mod-starter](https://github.com/mod-protocol/mod-starter) and the gateway if necessary
- [x] neynar api is a bit slow, consider using their mentions search api directly
- [x] publish package version

### CODE REVIEW

File: packages/farcaster/src/channels.ts [hunk 1]
```diff
@@ -1,8 +1,51 @@
...
+  id: "",
+  description: "followers",
+  name: "Home",
+  object: "channel",
+  parent_url: null,
+  image_url: "https://warpcast.com/~/channel-images/home.png",
+  channel_id: "home",
+} as const;
+
+export type VirtualChannel = typeof homeVirtualChannel;
```
Comment (stephancill) [inline packages/farcaster/src/channels.ts:23]: I think a generic `VirtualChannel` type would be useful for a reusable lighter representation of a channel e.g. for displaying in a list

**Review by stephancill — APPROVED** [review]

File: examples/api/src/app/api/farcaster/channels/v2/route.ts [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+      { message: "Something went wrong" },
+      { status: channelsRes.status }
+    );
+  }
+
+  const result = await channelsRes.json();
+
+  const channels = result.channels;
+
+  const channelsWithHome = hideHome
```
[the remark this answers is not in the document]
Comment (davidfurlong) [reply examples/api/src/app/api/farcaster/channels/v2/route.ts:34]: there are synthetic channels that you can't mention, such as /home

[the remark this answers is not in the document]
Comment (davidfurlong) [reply examples/api/src/app/api/farcaster/channels/v2/route.ts:34]: could do with a comment

**Comment by davidfurlong** [comment]
@stephancill should be good to go

**Review by stephancill — APPROVED** [review]
LGTM

**Pull request merged**